python/tchannel/thrift/scheme.py

```diff
@@ -28,7 +28,7 @@
 from .. import scheme
 
 
-class ThriftArgScheme(scheme.ArgScheme):
+class ThriftArgScheme(scheme.RawArgScheme):
```

Comment:
You would just compose in the parts you want to reuse. It's much cleaner that way, and the object hierarchy stays flat. You should avoid inheritance unless you need a contract to do type switching (polymorphism).